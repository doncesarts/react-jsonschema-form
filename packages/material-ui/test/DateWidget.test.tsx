import React from 'react';
import Form from "../src/index";
import { JSONSchema7 } from "json-schema";
import renderer from "react-test-renderer";

describe("DateWidget", () => {
  test("Render date input with title property defined", () => {
      const schema: JSONSchema7 = {
        type: "string", format: "date",  title: "date title" 
      };
      const tree = renderer.create(<Form schema={schema} />).toJSON();
      expect(tree).toMatchSnapshot();
  });
  test("Render date input without title property defined", () => {
      //The label of this input should be set based on the name of the property name, "dateNoTitle"
      const schema: JSONSchema7 = {
        type: "object",
        properties: {
          dateNoTitle: { type: "string", format: "date" }
        }
      };
      const tree = renderer.create(<Form schema={schema} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

