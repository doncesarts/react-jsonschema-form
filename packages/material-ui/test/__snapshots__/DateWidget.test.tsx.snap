// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateWidget Render date input with title property defined 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
          data-shrink={true}
          htmlFor="root"
          id="root-label"
        >
          date title
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="root"
            name="nodejs"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="date"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-info"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`DateWidget Render date input without title property defined 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <div
      className="MuiGrid-root makeStyles-root-57 MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        style={
          Object {
            "marginBottom": "10px",
          }
        }
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
                htmlFor="root_dateNoTitle"
                id="root_dateNoTitle-label"
              >
                dateNoTitle
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  id="root_dateNoTitle"
                  name="nodejs"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="date"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-info"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
